Xamarin.Forms.Core/ToolbarItem.cs

```diff
@@ -12,11 +12,34 @@ public class ToolbarItem : MenuItem
 
 		static readonly BindableProperty PriorityProperty = BindableProperty.Create("Priority", typeof(int), typeof(ToolbarItem), 0);
 
+		public static readonly BindableProperty IsVisibleProperty = BindableProperty.Create("IsVisible", typeof(bool), typeof(ToolbarItem), true, BindingMode.TwoWay, propertyChanged: (bindable, oldvalue, newvalue) =>
+		{
+			var item = bindable as ToolbarItem;
+
+			if (item != null)
+			{
+				if (item.Parent == null)
+					return;
+
+				var items = ((Page)item.Parent).ToolbarItems;
+
+				if ((bool)newvalue && !items.Contains(item))
+				{
+					Device.BeginInvokeOnMainThread(() => { items.Add(item); });
```

Comment:
I don't think adding/removing the toolbar items from the page is the best way to go about this. That seems like an unexpected side effect of setting visibility. 

Can we maybe use ToolbarTracker and filter out the visible items here?
https://github.com/xamarin/Xamarin.Forms/blob/bd31e1e9fc8b2f9ad94cc99e0c7ab058174821f3/Xamarin.Forms.Core/Internals/ToolbarTracker.cs#L50

Also if we got rid of Linq in that class that would be great as well.

Reply:
100% agree with @PureWeen here. handle that closer to the platform

Reply:
100% agree with @PureWeen, and we need make sure the position of item is still same as before when item become visible again.